### ISSUE TITLE
Support source:markdown

### ISSUE THREAD

**Issue by brentsimmons** [Ranchero-Software/NetNewsWire#3896]
`source:markdown` should be preferred, when present, to `content:encoded` and `description`. The parser should render the Markdown as HTML and set the already-existing property for HTML content on the parsed item. Since we don’t have a need to store the Markdown as Markdown, we won’t. This limits the changes to just the parser (in RSParser) — there shouldn’t be any change needed to app code.

The parser should also check the namespace and do that correctly — don’t assume the `source:` part will always be `source:` (despite the fact that we do do that with some other namespaces).

We’re not implementing the rest of `source` yet. We could some day, yes, but we’re starting with this because it’s the really cool part. (Goes to helping turn the web into a writing environment.)

Some links about this:

http://scripting.com/2022/07/18/163658.html?title=twitterRssMarkdown
http://scripting.com/2022/07/19/152235.html?title=devNotesForMarkdownInFeeds

PS Our decision to not store the Markdown text as Markdown makes sense right now, but it could change in the future. If so, we’ll write up a new ticket and do that work.

**Comment by scripting** [comment]
Thanks Brent! This is great news. RSS and Markdown have always imho belonged together. ;-)

**Comment by Wevah** [comment]
http://feedland.org/twitterfeeds/davewiner/rss.xml is a 404 now. :(

**Comment by scripting** [comment]
It should be --

http://data.feedland.org/twitterfeeds/davewiner/rss.xml

Where did you see a reference to it?

**Comment by Wevah** [comment]
> Where did you see a reference to it?

http://scripting.com/2022/07/19/152235.html?title=devNotesForMarkdownInFeeds, bullet point 5 (To get an idea how that works, have a look at the [RSS feed](http://xmlviewer.scripting.com/?url=http://feedland.org/twitterfeeds/davewiner/rss.xml) for my Twitter account. […])

**Comment by Wevah** [comment]
(I'm poking around at this a little, for fun, but if anyone else decides to jump on first it that's okay too!)

**Comment by brentsimmons** [comment]
To render Markdown as HTML: we will need to add Apple’s swift-markdown package, and write code to convert the AST (parsed Markdown data structure) to HTML. This should be fairly straightforward.

I’m thinking the parser should add a markdown field to the article objects it generates. At save-in-database time, that’s when Markdown rendering will happen (on a background thread). The rendered Markdown will be saved in the database as the main content. (Markdown will not be preserved — we don’t need it for anything.)

Of course we’ll have to handle edge cases like where Markdown exists but is an empty string and the normal HTML content is non-empty.

**Comment by brentsimmons** [comment]
Done. I ended up storing the markdown in the articles database — as opposed to just rendering it as HTML — so that NetNewsWire could take part in some kind of future editing flow. (But I’m not thinking right now about what that flow would look like.)

**Comment by scripting** [comment]
@brentsimmons -- does this mean that the next version of NNW will support source:markdown?

**Comment by brentsimmons** [comment]
Yes — the next version, 6.2, will support source:markdown. It should be out probably next week. In testing now.

**Comment by gingerbeardman** [comment]
Are there any example feeds in the wild where we can see source:markdown?

**Comment by brentsimmons** [comment]
@gingerbeardman Scripting News: http://scripting.com/rss.xml

Micro.blog feeds now include Markdown too. Manton Reece’s feed, for instance: https://www.manton.org/feed.xml

(I haven’t added Markdown to my blog’s feed yet, but I plan to.)
